python/GafferImageUI/ImageInspector.py

```diff
@@ -0,0 +1,460 @@
+		# thread-safety, because the PathListingWidget will access the context
+		# on a background thread.
+		context = Gaffer.Context( self.getContext() )
+		self.__formatPathListing.setPath(
+			_FormatPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+		self.__channelsPathListing.setPath(
+			_ChannelPath( self.scriptNode(), self.__settingsNode["__imageStats"]["in"], context, "/" ),
+		)
+		self.__metadataPathListing.setPath(
+			_MetadataPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+
+GafferUI.Editor.registerType( "ImageInspector", ImageInspector )
+
+##########################################################################
+# Metadata controlling the settings UI
+##########################################################################
+
+Gaffer.Metadata.registerNode(
+
+	ImageInspector.Settings,
+
+	plugs = {
+
+		"view" : [
+
+			"plugValueWidget:type", "GafferImageUI.ViewPlugValueWidget",
+			"description", "The view to inspect",
+
+		],
+
+	}
+
+)
+
+##########################################################################
+# _ImagePath
+##########################################################################
+
+class _ImagePath( Gaffer.Path ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		Gaffer.Path.__init__( self, path, root, filter )
+
+		self.__scriptNode = scriptNode
+		self._context = context
+		self._image = image
+		self.__imagePlugDirtiedConnection = self._image.node().plugDirtiedSignal().connect(
+			Gaffer.WeakMethod( self.__imagePlugDirtied, fallbackResult = None ), scoped = True
+		)
+
+	def copy( self ) :
+
+		return self.__class__( self.__scriptNode, self._image, self._context, self[:], self.root(), self.getFilter() )
+
+	def cancellationSubject( self ) :
+
+		## \todo We really just want to return `self._image` here, but we
+		# can't because BackgroundTask can't find the right ScriptNode from it.
+		return self.__scriptNode["fileName"]
+
+	def _children( self, canceller ) :
+
+		return [
+			self.__class__( self.__scriptNode, self._image, self._context, self[:] + [ name ], self.root(), self.getFilter() )
+			for name in self._childNames( canceller )
+		]
+
+	def _childNames( self, canceller ) :
+
+		raise NotImplementedError
+
+	def __imagePlugDirtied( self, plug ) :
+
+		if plug.parent() == self._image and self._imagePlugAffectsPath( plug ) :
+			self._emitPathChanged()
+
+##########################################################################
+# _FormatPath
+##########################################################################
+
+class _FormatPath( _ImagePath ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		_ImagePath.__init__( self, scriptNode, image, context, path, root, filter )
+
+	def propertyNames( self, canceller = None ) :
+
+		return Gaffer.Path.propertyNames( self ) + [ "format:value" ]
```

Comment:
Is that a way to call the base class? What would be the difference with calling `super( _FormatPath, self ).propertyNames()`

Reply:
Yes, and I don't think there is any difference with `super()` in this case. I never really bought into `super` before, since it's more longwinded and I'm used to explicitly specifying the base class in C++ anyway. There might be an argument for adopting it now that Python 3 allows us to omit the arguments, but that's not a topic for this PR.